src/screens/profile-screen/utils.ts

```diff
@@ -0,0 +1,71 @@
+import { useMemo } from 'react'
+
+import { ID } from 'audius-client/src/common/models/Identifiers'
+import { getAccountUser } from 'audius-client/src/common/store/account/selectors'
+import { ProfileUser } from 'audius-client/src/common/store/pages/profile/types'
+import {
+  badgeTiers,
+  makeGetTierAndVerifiedForUser
+} from 'audius-client/src/components/user-badges/utils'
+
+import { useSelectorWeb } from 'app/hooks/useSelectorWeb'
+
+import { MIN_COLLECTIBLES_TIER } from './constants'
+
+/**
+ * Wraps our reselect tier selector in useMemo and useSelector
+ * to be safe for use in multiple components
+ *
+ * This was copied over from audius-client and useSelector was replaced
+ * with useSelectorWeb.
+ */
+export const useSelectTierInfo = (userId: ID) => {
+  const tierAndVerifiedSelector = useMemo(makeGetTierAndVerifiedForUser, [])
+  const res = useSelectorWeb(state => {
+    return tierAndVerifiedSelector(state as any, { userId })
+  })
+  return res
+}
+
+/**
+ *
+ * Encapsulates complex logic for determining whether we should show the
+ * profile's collectibles tab.
+ *
+ * User needs to be at a specific tier, needs to have collectibles,
+ * they need to be ordered, and they need to load correctly.
+ *
+ * If the user is viewing their own profile, the collectibles don't
+ * need to be ordered
+ */
+export const useShouldShowCollectiblesTab = (profile: ProfileUser) => {
```

Comment:
IMO this is the kind of logic that should be shared in common, feels weird to maintain this business logic twice

Reply:
yeah i def agree, mind if combine that logic at a later point?

Reply:
Sounds good!